plugins/org.wso2.developerstudio.eclipse.gmf.esb.persistence/src/org/wso2/developerstudio/eclipse/gmf/esb/internal/persistence/IterateMediatorTransformer.java

```diff
@@ -110,8 +111,14 @@ public void transformWithinSequence(TransformationInfo information,
 			
 			iterateMediator.setContinueParent(visualIterate.isContinueParent());
 			
-			iterateMediator.setId(visualIterate.getIterateID());	
-			
+			/*
+			 * Do not serialize when Iterate ID is empty/null or contain
+			 * whitespace
+			 */
+			if (StringUtils.isNotBlank(visualIterate.getIterateID())) {
+				iterateMediator.setId(visualIterate.getIterateID());
```

Comment:
Formatting issue is fixed in this PR : https://github.com/wso2/devstudio-tooling-esb/pull/272